### ISSUE TITLE
Strategus v1.0 - make negative control outcome shared resource optional

### ISSUE THREAD

**Issue by egillax** [OHDSI/Strategus#153]
Current implementation of CohortGeneratorModule in branch `v1.0-main` seems to require negative control outcome shared resource to exist. Which is something we don't need in prediction. Is it possible to make it optional?

**Comment by anthonysena** [comment]
Yes, that's a bug - I'll make sure that's an optional part of the analysis specification.

**Comment by anthonysena** [comment]
Providing a reprex of the problem

``` r
library(Strategus)
#> Loading required package: CohortGenerator
#> Loading required package: DatabaseConnector
#> Loading required package: R6
#> 
#> Attaching package: 'Strategus'
#> The following object is masked from 'package:CohortGenerator':
#> 
#>     uploadResults
```

``` r
cohortDefinitionSet <- getCohortDefinitionSet(
  settingsFileName = system.file("testdata/Cohorts.csv", package = "Strategus"),
  jsonFolder = system.file("testdata/cohorts", package = "Strategus"),
  sqlFolder = system.file("testdata/sql", package = "Strategus")
)
#> Loading cohortDefinitionSet
```

``` r
subsetOperations <- list(
  createDemographicSubset(
    name = "Demographic Criteria",
    ageMin = 18,
    ageMax = 64
  )
)
subsetDef <- createCohortSubsetDefinition(
  name = "test definition",
  definitionId = 1,
  subsetOperators = subsetOperations
)
cohortDefinitionSet <- cohortDefinitionSet |>
  addCohortSubsetDefinition(subsetDef)

# Create the settings
cgModuleSettingsCreator <- CohortGeneratorModule$new()
cohortSharedResourcesSpecifications <- cgModuleSettingsCreator$createCohortSharedResourceSpecifications(cohortDefinitionSet)
cgModuleSpecifications <- cgModuleSettingsCreator$createModuleSpecifications()

# Create analysis specification
analysisSpecifications <- createEmptyAnalysisSpecificiations() |>
  addSharedResources(cohortSharedResourcesSpecifications) |>
  addCohortGeneratorModuleSpecifications(cgModuleSpecifications)


connectionDetails <- Eunomia::getEunomiaConnectionDetails()
executionSettings <- Strategus::createCdmExecutionSettings(
  workDatabaseSchema = "main",
  cdmDatabaseSchema = "main",
  cohortTableNames = CohortGenerator::getCohortTableNames(cohortTable = "strategus_test"),
  workFolder = "temp/workFolder",
  resultsFolder = "temp/resultsFolder"
)

Strategus::execute(
  analysisSpecifications = analysisSpecifications,
  executionSettings = executionSettings,
  connectionDetails = connectionDetails
)
#> Currently in a tryCatch or withCallingHandlers block, so unable to add global calling handlers. ParallelLogger will not capture R messages, errors, and warnings, only explicit calls to ParallelLogger. (This message will not be shown again this R session)
#> Connecting using SQLite driver
#> EXECUTING: CohortGeneratorModule
#> ERROR: Negative control outcome shared resource not found!
#> 
#> -- EXECUTION SUMMARY -----------------------------------------------------------
#> x CohortGeneratorModule Negative control outcome shared resource not found! (Execution Time: 4.12 secs)
```

<sup>Created on 2024-08-19 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by anthonysena** [comment]
Changes now available on `v1.0-main`
